library/src/main/java/com/google/android/exoplayer/extractor/mp4/AtomParsers.java

```diff
@@ -1154,4 +1154,119 @@ public AvcCData(List<byte[]> initializationData, int nalUnitLengthFieldLength,
 
   }
 
+  /**
+   * Creates a {@link SampleSizeBox} for the stblAtom passed in.
+   * @param stblAtom the atom to read the sample size atoms from.
+   * @return an implementation of {@link SampleSizeBox} that can be used to parse sample information.
+   * @throws ParserException if no suitable {@link Atom}s are found for parsing.
+     */
+  public static SampleSizeBox sampleSizeBoxFor(Atom.ContainerAtom stblAtom) throws ParserException {
+      // Array of sample sizes.
+      final Atom.LeafAtom stszAtom = stblAtom.getLeafAtomOfType(Atom.TYPE_stsz);
+      final Atom.LeafAtom stz2Atom = stblAtom.getLeafAtomOfType(Atom.TYPE_stz2);
+
+      if (stszAtom == null && stz2Atom == null) {
+        throw new ParserException("Track has no sample table size information");
+      }
+
+    return stszAtom != null ? new StszSampleSizeBox(stszAtom) : new Stz2SampleSizeBox(stz2Atom);
+  }
+
+  /**
+   * Interface that allows us to abstract away the various implementations of sample size boxes. (IE: stsz and stz2)
+   */
+  interface SampleSizeBox {
+    /**
+     * @return the number of samples in this atom.
+     */
+    int getSampleCount();
+
+    /**
+     * @return the fixed sample size read from the atom. If 0, the atom has non-fixed sample sizes.
+     */
+    int getFixedSampleSize();
+
+    /**
+     * @param i the index in the sample size array we wish to get the size from.
+     * @return the size for the sample at the index passed.
+     */
+    int getSampleSizeAt(int i);
+  }
+
+  /**
+   * Reads sample sizes out of a 'stsz' atom.
+   */
+  static final class StszSampleSizeBox implements SampleSizeBox {
+    public final Atom.LeafAtom atom;
```

Comment:
Looks unused outside of the constructor. Hold a reference to the .data object directly? Ditto for class below.